servicetalk-loadbalancer/src/main/java/io/servicetalk/loadbalancer/DefaultHttpLoadBalancedConnection.java

```diff
@@ -0,0 +1,229 @@
+                                      final RootSwayingLeafRequestTracker tracker) {
+        this.responseErrorMapper = responseErrorMapper;
+        this.delegate = delegate;
+        this.tracker = tracker;
+        this.concurrencyController = concurrencyController;
+    }
+
+    @Override
+    public int score() {
+        return tracker.score();
+    }
+
+    @Override
+    public ReservedHttpConnection asConnection() {
+        return toReservedConnection(this, executionContext().executionStrategy());
+    }
+
+    @Override
+    public ReservedBlockingStreamingHttpConnection asBlockingStreamingConnection() {
+        return toReservedBlockingStreamingConnection(this, executionContext().executionStrategy());
+    }
+
+    @Override
+    public ReservedBlockingHttpConnection asBlockingConnection() {
+        return toReservedBlockingConnection(this, executionContext().executionStrategy());
+    }
+
+    @Override
+    public Completable releaseAsync() {
+        return concurrencyController.releaseAsync();
+    }
+
+    @Override
+    public Completable closeAsyncGracefully() {
+        return delegate.closeAsyncGracefully();
+    }
+
+    @Override
+    public Result tryRequest() {
+        return concurrencyController.tryRequest();
+    }
+
+    @Override
+    public boolean tryReserve() {
+        return concurrencyController.tryReserve();
+    }
+
+    @Override
+    public void requestFinished() {
+        concurrencyController.requestFinished();
+    }
+
+    @Override
+    public HttpConnectionContext connectionContext() {
+        return delegate.connectionContext();
+    }
+
+    @Override
+    public <T> Publisher<? extends T> transportEventStream(final HttpEventKey<T> eventKey) {
+        return delegate.transportEventStream(eventKey);
+    }
+
+    @Override
+    public Single<StreamingHttpResponse> request(final StreamingHttpRequest request) {
+        return delegate.request(request)
+                .liftSync(target -> new StreamingHttpResponseSubscriber(target, this));
+    }
+
+    @Override
+    public HttpExecutionContext executionContext() {
+        return delegate.executionContext();
+    }
+
+    @Override
+    public StreamingHttpResponseFactory httpResponseFactory() {
+        return delegate.httpResponseFactory();
+    }
+
+    @Override
+    public Completable onClose() {
+        return delegate.onClose();
+    }
+
+    @Override
+    public Completable onClosing() {
+        return delegate.onClosing();
+    }
+
+    @Override
+    public Completable closeAsync() {
+        return delegate.closeAsync();
+    }
+
+    @Override
+    public StreamingHttpRequest newRequest(final HttpRequestMethod method, final String requestTarget) {
+        return delegate.newRequest(method, requestTarget);
+    }
+
+    private static final class StreamingHttpResponseSubscriber implements Subscriber<StreamingHttpResponse> {
```

Comment:
@idelpivnitskiy pointed out that we're not fully tracking the response: we're only getting the head of the response. He showed me the `BeforeFinallyHttpOperator` which has the property of waiting until the body of the response has been received before finishing its operations, but it doesn't have a way to inspect the response such as in the case of a 5xx response. We don't have to make it perfect right now but we should keep it in mind at a minimum.